src/Simulator/IncrementalRedstoneSimulator/IncrementalRedstoneSimulator.cpp

```diff
@@ -0,0 +1,184 @@
+
+
+#include "Globals.h"
+
+#include "IncrementalRedstoneSimulator.h"
+#include "Chunk.h"
+
+#include "CommandBlockHandler.h"
+#include "DoorHandler.h"
+#include "RedstoneHandler.h"
+#include "RedstoneTorchHandler.h"
+#include "RedstoneWireHandler.h"
+#include "RedstoneRepeaterHandler.h"
+#include "RedstoneToggleHandler.h"
+#include "SolidBlockHandler.h"
+#include "RedstoneLampHandler.h"
+#include "RedstoneBlockHandler.h"
+#include "PistonHandler.h"
+#include "SmallGateHandler.h"
+#include "NoteBlockHandler.h"
+#include "TNTHandler.h"
+#include "PoweredRailHandler.h"
+#include "PressurePlateHandler.h"
+#include "TripwireHookHandler.h"
+#include "DropSpenserHandler.h"
+#include "RedstoneComparatorHandler.h"
+#include "TrappedChestHandler.h"
+
+
+
+
+
+std::unique_ptr<cRedstoneHandler> cIncrementalRedstoneSimulator::CreateComponent(cWorld & a_World, BLOCKTYPE a_BlockType, cIncrementalRedstoneSimulatorChunkData * a_Data)
+{
+	switch (a_BlockType)
+	{
+		case E_BLOCK_REDSTONE_TORCH_ON:
+		case E_BLOCK_REDSTONE_TORCH_OFF: return cpp14::make_unique<cRedstoneTorchHandler>(a_World);
+		case E_BLOCK_REDSTONE_LAMP_OFF:
+		case E_BLOCK_REDSTONE_LAMP_ON: return cpp14::make_unique<cRedstoneLampHandler>(a_World);
+		case E_BLOCK_REDSTONE_WIRE: return cpp14::make_unique<cRedstoneWireHandler>(a_World);
+		case E_BLOCK_REDSTONE_REPEATER_OFF:
+		case E_BLOCK_REDSTONE_REPEATER_ON: return cpp14::make_unique<cRedstoneRepeaterHandler>(a_World);
+		case E_BLOCK_WOODEN_BUTTON:
+		case E_BLOCK_STONE_BUTTON:
+		case E_BLOCK_LEVER: return cpp14::make_unique<cRedstoneToggleHandler>(a_World);
+		case E_BLOCK_STICKY_PISTON:
+		case E_BLOCK_PISTON: return cpp14::make_unique<cPistonHandler>(a_World);
+		case E_BLOCK_BLOCK_OF_REDSTONE: return cpp14::make_unique<cRedstoneBlockHandler>(a_World);
+		case E_BLOCK_TRAPDOOR:
+		case E_BLOCK_IRON_TRAPDOOR:
+		case E_BLOCK_FENCE_GATE: return cpp14::make_unique<cSmallGateHandler>(a_World);
+		case E_BLOCK_TNT: return cpp14::make_unique<cTNTHandler>(a_World);
+		case E_BLOCK_NOTE_BLOCK: return cpp14::make_unique<cNoteBlockHandler>(a_World);
+		case E_BLOCK_POWERED_RAIL:
+		case E_BLOCK_ACTIVATOR_RAIL:
+		case E_BLOCK_DETECTOR_RAIL: return cpp14::make_unique<cPoweredRailHandler>(a_World);
+		case E_BLOCK_COMMAND_BLOCK: return cpp14::make_unique<cCommandBlockHandler>(a_World);
+		case E_BLOCK_STONE_PRESSURE_PLATE:
+		case E_BLOCK_WOODEN_PRESSURE_PLATE:
+		case E_BLOCK_HEAVY_WEIGHTED_PRESSURE_PLATE:
+		case E_BLOCK_LIGHT_WEIGHTED_PRESSURE_PLATE: return cpp14::make_unique<cPressurePlateHandler>(a_World);
+		case E_BLOCK_TRIPWIRE_HOOK: return cpp14::make_unique<cTripwireHookHandler>(a_World);
+		case E_BLOCK_ACTIVE_COMPARATOR:
+		case E_BLOCK_INACTIVE_COMPARATOR: return cpp14::make_unique<cRedstoneComparatorHandler>(a_World);
+		case E_BLOCK_DROPPER:
+		case E_BLOCK_DISPENSER: return cpp14::make_unique<cDropSpenserHandler>(a_World);
+		case E_BLOCK_TRAPPED_CHEST: return cpp14::make_unique<cTrappedChestHandler>(a_World);
```

Comment:
Style! This switch is completely unreadable. Either sort it alphabetically and return in each case, or make each case a curly-brace block and add newlines to separate the multi-case blocks.